### ISSUE TITLE
DOSBox doesn't respond to Switch Pro controller on macOS

### ISSUE THREAD

**Issue by Ekamekia** [dosbox-staging/dosbox-staging#3992]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.82.0-rc1 (9d55e)

### What Operating System are you using?

macOS (Apple Silicon)

### If Other OS, please describe

MacOS 15.0.1

### Relevant hardware info

M1

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Dosbox won't respond to Nintendo Switch Pro controller on MacOS. Up until today I've been using a Playstation 4 controller and it has worked without any problems. Connecting the Switch controller is the same procedure as the Playstation controller, connect from the bluetooth menu and it shows up as 'Pro Controller'

I tested it out in OpenEmu first and it recognized the controller, but Dosbox completely ignores it, or is unaware of it both in games and in the mapper.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Obtain Pro Controller
2. Connect to Mac
3. Launch Dosbox
4. Enter Mapper
5. Nada

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by johnnovak** [comment]
Yeah, no idea. It's unlikely this will be addressed as I don't think any of us in the core team has this controller. Neither do we want to start adding controller specific hacks/workarounds/etc.

The real solution will be using SDL's ControllerDB project. Seems like they did add support for this particular controller a while ago:

https://github.com/libsdl-org/SDL/commit/404cb27677b4d6f953a55d181508b5f690df0993

Then whatever SDL and ControllerDB supports, we support it. But that won't happen anytime soon, so don't hold your breath. In the meantime, just use some "true and tested" controller with Staging (like the Xbox 360 controllers on Windows, they just work without a hitch).

**Comment by Ekamekia** [comment]
<<Btw, can does it work with Staging on Windows? Can you test that?>>

Sorry, Didn't see your question there. Tested it, Yes, it works in Windows.

**Comment by johnnovak** [comment]
> <<Btw, can does it work with Staging on Windows? Can you test that?>>
> 
> Sorry, Didn't see your question there. Tested it, Yes, it works in Windows.

Cheers. macOS specific SDL problem, I don't think we can do much about it. You could try other SDL2 apps on macOS that _don't_ use ControllerDB to confirm that (e.g., some other emulator).

**Comment by Ekamekia** [comment]
Yeah, I wouldn't worry about it at all. I'm probably the only asshat trying to use a _Switch_ Pro controller to play _DOS_ games on _MacOS_. Besides its started to have a massive input lag for whatever reason. Back to PS4!

**Comment by johnnovak** [comment]
As I thought, this was an SDL issue. [SDL 2.36.0](https://github.com/libsdl-org/SDL/releases/tag/release-2.32.6) was specifically released to fix these Switch controller issues on macOS:

> This is a stable bugfix release, with the following changes:
> - Fixed reliability of initializing Switch controllers on macOS
> - Fixed crash when controllers are disconnected on macOS

@Ekamekia We on SDL 2.36.0 in the dev build, so could you please confirm the issues has been fixed in our latest dev build? You can get it from [here](https://www.dosbox-staging.org/releases/development-builds/). The dev builds are not notarised, so you'll need to permit them to run in the security tab in System Settings after the first run attempt.

**Comment by johnnovak** [comment]
We're on SDL 2.36.0 which supposedly fixes this, but the user who reported this did not get back to us.

Closing as we can't do much about this other than to keep upgrading to the latest SDL2 version.
